framework/src/main/java/org/checkerframework/framework/type/AnnotatedTypeFactory.java

```diff
@@ -3028,9 +3028,10 @@ private AnnotationMirror getDeclAnnotation(
      * @param elt the element for which to determine annotations
      */
     public Set<AnnotationMirror> getDeclAnnotations(Element elt) {
-        if (cacheDeclAnnos.containsKey(elt)) {
```

Comment:
`if (map.containsKey(..) { ..}` is only equivalent to `value = map.get(...); if (value != null) {...}` if values in the map are never `null`.  Values in `cacheDeclAnnos` may be null, so this change should be reverted.  Can you check all the other changes to see if any of those maps contain null values?

Reply:
A way to be sure about that would be to run the Nullness Checker, though it is challenging to run on some parts of the codebase (I remember the parser and AST, in particular, causing problems in the past).

Reply:
@smillst I think the values in `cacheDeclAnnos` shouldn't be null because the only place where we put new values into it is also in this same function:

```
// Add the element and its annotations to the cache.
cacheDeclAnnos.put(elt, results);
```

But the `results` variable is always initialized by a call to `AnnotationUtils.createAnnotationSet();` which never returns null.